modules/contact/doc/content/source/constraints/ComputeWeightedGapCartesianLMMechanicalContact.md

```diff
@@ -44,6 +42,22 @@ gap is of order .01, then `c` should be set to 1e6 in order to bring
 components of the NCP function onto the same level and achieve optimal
 convergence in the non-linear solve.
 
+The `ComputeWeightedGapCartesianLMMechanicalContact` object computes the weighted gap and
+applies the KKT conditions using Lagrange multipliers defined in a global Cartesian reference frame.
+As a consequence, the number of contact constraints at each node will be two, in two-dimensional problems,
+and three, in three-dimensional problems. The normal contact pressure is obtained by projecting the Lagrange
+multiplier vector along the normal vector computed from the mortar generation objects. The result is a normal
+contact constraint, which, in general, will be a function of all (two or three) Cartesian Lagrange multipliers.
+This methodology only constrains one degree of freedom. The other degree(s) of freedom are constrained by
+enforcing that tangential tractions are identically zero. Note that, if friction with Cartesian Lagrange multipliers
+is chosen via [`ComputeFrictionalForceCartesianLMMechanicalContact`](/ComputeFrictionalForceCartesianLMMechanicalContact.md),
+those remaining nodal degrees of freedom are constraint using Coulomb constraints within a semi-smooth Newton approach. Usage of
+Cartesian Lagrange multipliers is recommended when condensing Lagrange multipliers via the variable condensation preconditioner
+(VCP) [`VariableCondensationPreconditioner`](/VariableCondensationPreconditioner.md).
+
+The user can also employed locally oriented Lagrange multipliers `ComputeWeightedGapLMMechanicalContact`](/ComputeWeightedGapLMMechanicalContact.md),
+which minimizes the number of contact constraints.
```

Comment:
Let me know if my understanding is correct or not... if you are using the Cartesian strategy, then you *must* have frictional contact constraints, correct? Or else you could potentially have a singular row if an axis is aligned with the mortar face?

On the other hand if you are not doing the Cartesian strategy, you can optionally choose to just use a single normal contact constraint. If you choose to add frictional constraints with a non-Cartesian strategy, then you should have the same number of contact constraints as the Cartesian strategy.

Is my analysis accurate or not?

Reply:
>  if you are using the Cartesian strategy, then you must have frictional contact constraints, correct? 

I think what you say in general is right. But when we go with the Cartesian strategy, we will _always_ have the same number of constraints, which coincides with the dimension of the problem. You can still choose to not have frictional constraints by just imposing that tangential tractions can be zero (i.e. we _have_ to add constraints but are frictionless), which is what happens when we don't add additional constraints for the Lagrange multipliers defined in the local frame with the normal constraint. So, using the `ComputeWeightedGapCartesianLMMechanicalContact` constraint allows the user to have Cartesial LMs and frictionless behavior.

Reply:
> On the other hand if you are not doing the Cartesian strategy, you can optionally choose to just use a single normal contact constraint. If you choose to add frictional constraints with a non-Cartesian strategy, then you should have the same number of contact constraints as the Cartesian strategy.

👍

Reply:
> Or else you could potentially have a singular row if an axis is aligned with the mortar face?

There is a strategy in the contact constraint to handle the degree of freedom associated with each of the constraint. To avoid, for example, associate a normal constraint along x with a z Lagrange multiplier. This has shown to work in examples in which two blocks in contact are rotated from 0 to 90 degrees (@dewenyushu checked this), but it is intuitively to think the structure of the Jacobian will be poorer with this approach compared to the local LM approach. Sorry for the choppy answer.

Reply:
So if you want to do frictional contact with the Cartesian strategy, then in the MOOSE implementation speak you have a single contact constraint, and it is the frictional object. This single frictional object acts on the all the LMs (2 in 2D, 3 in 3D). If you want frictionless, then you use the frictionless object (`*WeightedGap*`); again this single constraint object acts on all the LMs.

This is different from the local LM approach in which we have a constraint object per Lagrange Multiplier (one weighted gap constraint object for the normal LM and 1-2 frictional constraint objects for the tangential LMs).

That about right?

Reply:
In the local LM approach, we have one single object for the frictional forces which may add one or two constraints depending on whether we are doing 2D or 3D. 

On the other hand, if we were speaking of adding the generalized forces corresponding to the contact constraint, with the Cartesian LM approach, we only need one object (`CartesianMortarMechanicalContact`) because there is no need to project the contribution of the LM, since it's already in the global frame.

Reply:
> In the local LM approach, we have one single object for the frictional forces which may add one or two constraints depending on whether we are doing 2D or 3D.

Oh you are right.

> On the other hand, if we were speaking of adding the generalized forces corresponding to the contact constraint, with the Cartesian LM approach, we only need one object (`CartesianMortarMechanicalContact`) because there is no need to project the contribution of the LM, since it's already in the global frame.

That's nice!